pyirf/interpolation/component_estimators.py

```diff
@@ -212,31 +217,35 @@ def __init__(
             grid_points,
         )
 
-        if not isinstance(bin_edges, np.ndarray):
-            raise TypeError("Input bin_edges is not a numpy array.")
-        elif not isinstance(bin_contents, np.ndarray):
+        if not isinstance(bin_contents, np.ndarray):
             raise TypeError("Input bin_contents is not a numpy array.")
-        elif bin_contents.shape[-1] != (bin_edges.shape[0] - 1):
-            raise ValueError(
-                f"Shape missmatch, bin_edges ({bin_edges.shape[0] - 1} bins) "
-                f"and bin_contents ({bin_contents.shape[-1]} bins) not matching."
-            )
         elif self.n_points != bin_contents.shape[0]:
             raise ValueError(
                 f"Shape missmatch, number of grid_points ({self.n_points}) and "
                 f"number of histograms in bin_contents ({bin_contents.shape[0]}) "
                 "not matching."
             )
+        elif not isinstance(bin_edges, np.ndarray):
+            raise TypeError("Input bin_edges is not a numpy array.")
+        elif bin_contents.shape[-1] != (bin_edges.shape[0] - 1):
+            raise ValueError(
+                f"Shape missmatch, bin_edges ({bin_edges.shape[0] - 1} bins) "
+                f"and bin_contents ({bin_contents.shape[-1]} bins) not matching."
+            )
 
         if interpolator_kwargs is None:
             interpolator_kwargs = {}
 
         if extrapolator_kwargs is None:
             extrapolator_kwargs = {}
 
-        if not issubclass(interpolator_cls, DiscretePDFInterpolator):
+        if not (
+            issubclass(interpolator_cls, DiscretePDFInterpolator)
+            or issubclass(interpolator_cls, DiscretePDFNearestNeighborSearcher)
```

Comment:
you can [register](https://docs.python.org/3/library/abc.html#abc.ABCMeta.register) a class as   a virtual subclass of an abstract base class. I think this is appropriate here, the neighbor classes implement the correct API without actually inheriting.

THis would leave the `component_estimator` classes completely unchanged/